### ISSUE TITLE
Version name uniqueness

### ISSUE THREAD

**Issue by bodom0015** [whole-tale/wt_versioning#8]
**Steps to Reproduce**
1. Navigate to https://girder.local.wholetale.org/api/v1#!/version/version_create
2. Using Swagger UI, create a Tale (if you haven't already) and then grab its ID
3. Use the `taleId` to create a `version` with the name: `name`
4. With `force=true`, attempt to create a version with the same `name` again with the same `taleId`

**Expected Results**
Version should either be created successfully, or a 400-level error should be thrown. 400-level HTTP errors indicate that the client has sent invalid data that will need to be adjusted before re-submitting the request.

*Aside:* Do we need to enforce uniqueness on the name? The name isn't the primary key in the database, and the combination of a name + timestamp should still allow a user to determine which version is which if duplicate names are present. Enforcing unique names here will complicate the validation logic a bit, as we currently have no lookup for version by name - we could maybe reuse the `GET /version` + filtering parameters for this, if this behavior is truly desirable, but I would opt for adding an endpoint like `GET /version/{name}` or `GET /version/{name}/exists` in that case.

**Actual Results**
HTTP Status code reports `500`, but stack trace indicates that this should maybe be `400`? 
```
{
  "message": "ValueError: ValueError('Name already exists: name', 400)",
  "trace": [
    "<FrameSummary file /girder/girder/api/rest.py, line 630 in endpointDecorator>",
    "<FrameSummary file /girder/girder/api/rest.py, line 1224 in POST>",
    "<FrameSummary file /girder/girder/api/rest.py, line 967 in handleRoute>",
    "<FrameSummary file /girder/girder/api/access.py, line 63 in wrapped>",
    "<FrameSummary file /girder/girder/api/rest.py, line 445 in wrapped>",
    "<FrameSummary file /girder/girder/api/describe.py, line 709 in wrapped>",
    "<FrameSummary file /girder/plugins/wt_versioning/server/resources/version.py, line 126 in create>",
    "<FrameSummary file /girder/plugins/wt_versioning/server/resources/abstract_resource.py, line 53 in _checkNameSanity>"
  ],
  "type": "internal"
}
```

500-level HTTP errors indicate a server-side problem that the user should not resubmit, as they typically cannot be fixed by adjusting the data that the client is sending.

### TL;DR
I think the API is returning the wrong status code when submitting a duplicate version name, but am also curious why these names need to be unique at all

**Comment by hategan** [comment]
Since we want to map versions to directories, the names need to be unique.

I'll fix the exception. The code is wrongly using ValueError instead of RestException.

**Comment by bodom0015** [comment]
That makes sense - I had not thought about the fact that they map to directories on disk.

In other places (e.g. Tale workspaces), we have previously used the resource ID as the directory name to guarantee uniqueness.

**Comment by hategan** [comment]
> In other places (e.g. Tale workspaces), we have previously used the resource ID as the directory name to guarantee uniqueness.

We still want the filesystem to be usable as a user interface for this. The FS implementation supports all the operations that the UI does, such as renaming, deleting, etc. Using IDs as the directory name would make that difficult if not impossible (renaming would not make sense).
